src/main/java/com/hackclub/hccore/listeners/PlayerListener.java

```diff
@@ -96,6 +96,20 @@ public void onAsyncPlayerChat(final AsyncPlayerChatEvent event) {
   public void onPlayerJoin(final PlayerJoinEvent event) {
     Player player = event.getPlayer();
     this.plugin.getDataManager().registerPlayer(player);
+    PlayerData data = this.plugin.getDataManager().getData(player);
+
+    // Check for slack link
+    if (data == null || data.getSlackId() == null) {
```

Comment:
needs to also check to ensure that A: slack is enabled B: a new config options enforce-slack-link is enabled.
(name pending)